### PR TITLE
Added types to Constants

### PR DESCRIPTION
**Pull request by EvanBacon**
* Updated UIWebView to WKWebView
* Added missing web props

### CODE REVIEW

**Review by ide — CHANGES_REQUESTED** [review]
Thank you for going through the constants and writing a type definition for it - think this will be helpful.

File: packages/expo-constants/ios/EXConstants/EXConstants.m [hunk 1]
```diff
@@ -40,10 +42,21 @@ - (NSDictionary *)constantsToExport
     __strong EXConstants *strongSelf = weakSelf;
     if (strongSelf) {
       if (!strongSelf.webViewUserAgent) {
-        UIWebView *webView = [[UIWebView alloc] init];
-        strongSelf.webViewUserAgent = [webView stringByEvaluatingJavaScriptFromString:@"navigator.userAgent"];
+        // We need to retain the webview because it runs an async task.
+        strongSelf->webView = [[WKWebView alloc] init];
```
[the remark this answers is not in the document]
Comment (EvanBacon) [reply packages/expo-constants/ios/EXConstants/EXConstants.m]: nah if you use a local reference it throws an error in the returned value.

File: packages/expo-constants/src/Constants.types.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    tool?: string;
+  };
+  bundleUrl?: string;
+  debuggerHost?: string;
+  mainModuleName?: string;
+  logUrl?: string;
+  [key: string]: any;
+}
+
+export interface PlatformInterface {
```
[the remark this answers is not in the document]
Comment (EvanBacon) [reply packages/expo-constants/src/Constants.types.ts]: lol .. It's like the part of constants that you interface with tho. Will change it though

File: packages/expo-constants/ios/EXConstants/EXConstants.m [hunk 3]
```diff
@@ -3,9 +3,11 @@
 #import <EXConstants/EXConstants.h>
 #import <EXConstantsInterface/EXConstantsInterface.h>
 
-#import <UIKit/UIWebView.h>
+#import <WebKit/WKWebView.h>
 
-@interface EXConstants ()
+@interface EXConstants () {
+  WKWebView *webView;
```
[the remark this answers is not in the document]
Comment (EvanBacon) [reply packages/expo-constants/ios/EXConstants/EXConstants.m]: can't, maybe I should leave a comment so no one tries to remove it.

**Review by ide — APPROVED** [review]

**Comment by ide** [comment]
LGTM

**Pull request merged**